src/osd/OSDMap.h

```diff
@@ -1165,6 +1165,26 @@ class OSDMap {
     }
     return false;
   }
+  bool get_primary_shard(const pg_t& pgid, int *primary, spg_t *out) const {
+    auto i = get_pools().find(pgid.pool());
+    if (i == get_pools().end()) {
+      return false;
+    }
+    vector<int> acting;
+    pg_to_acting_osds(pgid, &acting, primary);
+    if (i->second.is_erasure()) {
+      for (uint8_t i = 0; i < acting.size(); ++i) {
+	if (acting[i] == *primary) {
+	  *out = spg_t(pgid, shard_id_t(i));
+	  return true;
+	}
+      }
+    } else {
```

Comment:
Fine as-written, but it also could be a check against the inverse above, in which case the branches can be removed from the function, since this returns.